### PR TITLE
Storage Write API Basic mode integration tests

### PR DESCRIPTION
**Pull request by b-goyal**
Addresses CC-19648
Includes -
1. Integration test to verify basic flow with Json type
2. Integration test to verify basic flow with Avro SR based type
3. Integration test to verify Schema updates
4. Integration test to verify auto Table creation
5. Integration test to verify correct DLQ routing
6. Integration test to verify stream creation

### CODE REVIEW

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQueryErrantRecordHandlerIT.java [hunk 1]
```diff
@@ -73,6 +74,98 @@ public void close() throws Exception {
...
+
+    // Instantiate the converters we'll use to send records to the connector
+    converter = new AvroConverter();
+    converter.configure(Collections.singletonMap(
+                    SCHEMA_REGISTRY_URL_CONFIG,schemaRegistryUrl
+            ), false
+    );
+
+    List<SchemaAndValue> records = getRecords();
+    schemaRegistry.produceRecords(converter, records, topic);
```
Comment (aniketshrimal) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQueryErrantRecordHandlerIT.java:99]: Minor: lets add a comment here - where we mention we are generating invalid records which leads to INVALID_ARGUMENT error.

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQueryStorageWriteApiSinkConnectorIT.java [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            testRows = readAllRows(bigQuery, table, "f3");
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        assertEquals(expectedRows(), testRows.stream().map(row -> row.get(0)).collect(Collectors.toSet()));
+    }
+
+    @Test
+    public void testBaseAvrodWithSchema() throws InterruptedException {
```
Comment (aniketshrimal) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQueryStorageWriteApiSinkConnectorIT.java:184]:
```suggestion
    public void testBaseAvroWithSchema() throws InterruptedException {
```

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java [hunk 3]
```diff
@@ -42,6 +45,16 @@ public void setup() throws Exception {
...
+            // Try to wait for 5 minutes if table is seen.
+            int attempts = 10;
+            while(bigQueryInstance.getTable(tableId) == null && attempts > 0)  {
+                logger.debug("Busy waiting for table {} to appear! Attempt {}", tableId.getTable(), (10-attempts));
+                Thread.sleep(TimeUnit.SECONDS.toMillis(30));
+                attempts--;
+            }
+            if(attempts == 0) {
+                logger.error("Created table is not available");
+            }
```
Comment (aniketshrimal) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java:57]: What is the reason behind adding this change? Does it reduce the flakiness?

**Review by aniketshrimal — COMMENTED** [review]
Overall LGTM! Nice test coverage. Have left some minor comments.

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQueryErrantRecordHandlerIT.java (hunk 1 above)
Comment (b-goyal) [reply kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQueryErrantRecordHandlerIT.java:99]: Nice suggestion, adding.

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java [hunk 4]
```diff
@@ -42,6 +45,16 @@ public void setup() throws Exception {
...
             // table takes time after creation before being available for operations. You may have to wait a few minutes (~5 minutes)
+            // Try to wait for 5 minutes if table is seen.
+            int attempts = 10;
+            while(bigQueryInstance.getTable(tableId) == null && attempts > 0)  {
+                logger.debug("Busy waiting for table {} to appear! Attempt {}", tableId.getTable(), (10-attempts));
+                Thread.sleep(TimeUnit.SECONDS.toMillis(30));
+                attempts--;
+            }
+            if(attempts == 0) {
+                logger.error("Created table is not available");
```
Comment (pbadani) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java:56]: do we need to throw an assertion error here?

**Review by pbadani — APPROVED** [review]
LGTM with one comment.

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java (hunk 4 above)
Comment (b-goyal) [reply kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java:56]: Should be good to add that

**Review by aniketshrimal — APPROVED** [review]

**Pull request merged**